src/main/scala/net/snowflake/spark/snowflake/SnowflakeTelemetry.scala

```diff
@@ -100,7 +103,13 @@ object SnowflakeTelemetry {
       metric.put(TelemetryOOBFields.EXCEPTION_STACKTRACE, "NA")
     }
 
-    oobTelemetryService
+    // The constructor of TelemetryEvent.LogBuilder uses
+    // TelemetryService.getInstance to retrieve the thread safe instance
+    // and use it. So getOobTelemetryService() needs to be called before
+    // creating the LogBuilder object to make sure TelemetryService has
+    // been setup.
+    TelemetryService.enable()
```

Comment:
Why do you want to call this repeatedly ? Move this out into the initialization of the class.